swapi/src/schema/connections.js

```diff
@@ -32,19 +37,46 @@ export function connectionFromUrls(
   prop: string,
   type: GraphQLOutputType
 ): GraphQLFieldConfig {
-  var {connectionType} = connectionDefinitions({name: name, nodeType: type});
+  var {connectionType} = connectionDefinitions({
+    name: name,
+    nodeType: type,
+    connectionFields: () => ({
+      totalCount: {
+        type: GraphQLInt,
+        resolve: (conn) => conn.totalCount,
+        description:
+`A count of the total number of objects in this connection, ignoring pagination.
+This allows a client to fetch the first five objects by passing "5" as the
+argument to "first", then fetch the total count so it could display "5 of 83",
+for example.`
+      },
+      // $FlowIssue Computed propertes
+      [prop]: {
+        type: new GraphQLList(type),
+        resolve: (conn) => conn.edges.map((edge) => edge.node),
+        description:
+`A list of all of the objects returned in the connection. This is a convinience
+field provided for quickly exploring the API; rather than querying for
+"{ edges { node } }" when no edge data is needed, this field can be be used
+instead. Note that when clients like Relay need to fetch the "cursor" field on
+the edge to enable efficient pagination, this shortcut cannot be used, and the
+full "{ edges { node } }" version should be used instead.`
+      }
+    })
+  });
   return {
     type: connectionType,
     args: connectionArgs,
-    resolve: (obj, args) => {
+    resolve: async (obj, args) => {
       var promisedArray = [];
       if (obj[prop]) {
         promisedArray = obj[prop].map(getObjectFromUrl);
       }
-      return connectionFromPromisedArray(
-        Promise.all(promisedArray),
-        args
-      );
+      var array = await Promise.all(promisedArray);
+      return {
+        ...connectionFromArray(array, args),
+        totalCount: array.length
```

Comment:
This is for the case where we're converting a list of URLs into a connection, so we're just counting the URLs here.